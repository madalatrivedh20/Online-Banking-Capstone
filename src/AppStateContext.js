import { createContext, useState, useContext } from "react";

const AppStateContext = createContext();

export const AppStateProvider = ({ children }) => {
  const [isAuthenticated, setIsAuthenticated] = useState(false);
  const [issocialAuthenticated, setIssocialAuthenticated] = useState(false);
  const [user, setUser] = useState({});
  const [allUsers, setAllUsers] = useState([]);

  const value = {
    isAuthenticated,
    setIsAuthenticated,
    user,
    issocialAuthenticated,
    setIssocialAuthenticated,
    setUser,
    allUsers,
    setAllUsers
  };
  return <AppStateContext.Provider value={value}>{children}</AppStateContext.Provider>;
};

const useAppState = () => {
  const context = useContext(AppStateContext);

  if (context === undefined) {
    throw new Error("useAppState must be used within AppStateContext");
  }

  return context;
};

export default useAppState;
