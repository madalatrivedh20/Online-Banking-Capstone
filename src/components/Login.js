//Importing all the necessary modules

import '../style/Login.css';
import { React, useState, useEffect } from 'react';
import { toast } from 'react-toastify';
import { useNavigate } from 'react-router-dom';
import validator from 'validator';
import pic from '../style/download.png';
import { initializeApp } from "firebase/app";
import { getAuth, GoogleAuthProvider, signInWithPopup, FacebookAuthProvider } from "firebase/auth";
import { getAnalytics } from "firebase/analytics";
import useAppContext from '../AppStateContext';
import GoogleButton from "react-google-button"


const FormHeader = props => (
  <div>
    <div className='imgcontainer'>
      <img src={pic} alt="Avatar" className="avatar"></img>
    </div>
    <h2 id="headerTitle">{props.title}</h2>
  </div>
);

function Login() {

  const { setIsAuthenticated, setUser, setIssocialAuthenticated,allUsers } = useAppContext();

  const navigate = useNavigate();

  const [data, setData] = useState({
    email: "",
    password: ""
  });


  const changeHandler = e => {
    setData({ ...data, [e.target.name]: [e.target.value] });

  };

  const firebaseConfig = {
    apiKey: "AIzaSyAQaUwSVM1OSQoh5uS7qiniQO4beSNb1JM",
    authDomain: "online-banking-2.firebaseapp.com",
    projectId: "online-banking-2",
    storageBucket: "online-banking-2.appspot.com",
    messagingSenderId: "647396172156",
    appId: "1:647396172156:web:d876621f4a8bb6b12744f9",
    measurementId: "G-8VLQ6LQBVN"
  };

  // Initialize Firebase
  const app = initializeApp(firebaseConfig);
  const analytics = getAnalytics(app);
  const auth = getAuth(app);

  const go_provider = new GoogleAuthProvider();
  const fb_provider = new FacebookAuthProvider();

  //Function invoked on sign-in of google
  const signInWithGoogle = () => {
    signInWithPopup(auth, go_provider)
      .then((result) => {
        const name = result.user.displayName;
        const email = result.user.email;
        const profilePic = result.user.photoURL;

        setIssocialAuthenticated(true);
        toast.success("Welcome", name);
        navigate('/')
          ;
      })
      .catch((error) => {
        console.log(error);
      });
  };


  const submitHandler = e => {

    // Validates email and checks if the password and email of the user matches
    e.preventDefault();
    let email_str = data['email'][0];
    let password_str = data['password'][0];
    let checkemail = (validator.isEmail(email_str));

    function checkemailpassofuser(obj) {
      console.log("entered checkemail")
      console.log(obj)
      let flag = 0;
      if (obj.email === email_str && obj.password === password_str) {
        flag = 1;
      }
      return flag;


    }
    if (!checkemail) {
      toast.error("Please enter a valid e-mail");

    }

    if (checkemail) {
      
      const result = allUsers.filter(checkemailpassofuser);

      if (result[0]) {

        const { id, accno, acctype } = result[0];
        setUser({ id, accno, acctype });
        setIsAuthenticated(true);
        toast.success("Logged In successfully");
        navigate('/balance');

      }
      else {
        toast.error("Wrong credentials");

      }
    }
  };

  return (
    <div className="forms">
      <FormHeader title="Login" />
      <div>
        <form onSubmit={submitHandler}>
          <div className="row">
            <label>Email</label>
            <input autofocus required type="text" name="email" placeholder="Enter your Email" value={data.email} onChange={changeHandler} />
          </div>
          <div className="row">
            <label>Password</label>
            <input required type="password" name="password" placeholder="Enter your Password" value={data.password} onChange={changeHandler} />
          </div>
          <div id="button" className="row">
            <button>Log In</button>

          </div>
          <hr />
          <div id="button" className="row">
            <div style={{ "textAlign": "center", "marginLeft": "0px" }}>
              <ul>
                <GoogleButton onClick={signInWithGoogle} />


              </ul>
            </div>



          </div>
        </form>

      </div>



      <div className="container" style={{ "backgroundColor": "#f1f1f1", "height": "50px", "textAlign": "center" }}>
        New User ? <a href="/register">Register Here</a>
      </div>


    </div>
  );
}

export default Login;



