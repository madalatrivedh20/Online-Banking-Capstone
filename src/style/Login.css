@import url('https://fonts.googleapis.com/css?family=Open+Sans:600|Work+Sans:400,500,600,700&display=swap');

body {
  background: rgb(255, 4, 0);
  background: linear-gradient(
    135deg,
    rgb(255, 19, 2) 0%,
    rgb(232, 190, 1) 100%
  );
  font-family: 'work sans';
}

* {
  box-sizing: border-box;
}

/* Create two unequal columns that floats next to each other */
.column {
  float: left;
  padding: 10px;
  height: 300px; /* Should be removed. Only for demonstration */
}
.columns {
  display: grid;
  grid-template-columns: 50% 50%;
}

.left {
  width: 25%;
}

.right {
  width: 75%;
}

/* Clear floats after the columns */
.row:after {
  content: '';
  display: table;
  clear: both;
}
.fa-twitter {
  background: #55acee;
  color: white;
}
.fa {
  padding: 20px;
  font-size: 30px;
  width: 60px;
  height: 60px;
  text-align: left;
  text-decoration: none;
  margin: 5px 2px;
  border-radius: 5px;
}

.fa:hover {
  opacity: 0.7;
}

.fa-facebook {
  background: #3b5998;
  color: white;
}

.fa-twitter {
  background: #55acee;
  color: white;
}

.fa-google {
  background: #dd4b39;
  color: white;
}
html {
  font-size: 80%;
}
.container {
  padding: 16px;
  padding-top: 20px;
}
.forms {
  max-width: 500px;
  min-width: 300px;
  max-height: 800px;
  width: 800px;
  height: 1000%;
  margin: 100px auto;
  background-color: #ffffff;
  border-radius: 25px;
}
#message {
  text-align: center;
  width: 60%;
  height: 0%;

  padding: 0;
  display: flex;
  justify-content: center;
  font-size: 10px;
}

.imgcontainer {
  text-align: center;
  margin: 24px 0 12px 0;
  position: relative;
  padding-top: 30px;
}

img.avatar {
  width: 40%;
  border-radius: 50%;
}

#headerTitle {
  text-align: center;
  font-family: 'open sans', sans-serif;
  padding: 2rem 0;
  margin: 0;
  font-size: 2rem;
}

.row {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: 2rem;
  max-width: 100%;
}

.row input {
  width: 80%;
  box-sizing: border-box;
  border: none;
  font-size: 1.3rem;
  padding-left: 1 rem;
  padding-bottom: 1rem;
  box-shadow: inset 0px -3px 0px 0px rgba(187, 187, 187, 0.2);
  transition: box-shadow 0.2s ease-in;
}

.row select {
  width: 80%;
  box-sizing: border-box;
  border: none;
  font-size: 1.3rem;
  padding-left: 1 rem;
  padding-bottom: 1rem;
  box-shadow: inset 0px -3px 0px 0px rgba(187, 187, 187, 0.2);
  transition: box-shadow 0.2s ease-in;
}

.row input:focus {
  box-shadow: inset 0px -3px 0px 0px rgba(34, 193, 195, 0.7);
  outline: none;
}

.row input::-webkit-input-placeholder {
  opacity: 1;
  transition: opacity 0.25s ease-out;
}

.row input:hover::-webkit-input-placeholder,
.row input:focus::-webkit-input-placeholder {
  opacity: 0;
}

.row label {
  align-self: start;
  padding-left: 4.5rem;
  padding-bottom: 0.5rem;
  color: rgba(187, 187, 187, 0.9);
}

.row button {
  border-radius: 25px;
  width: 80%;
  height: 40px;
  font-size: 1.3rem;
  color: white;
  font-weight: 700;
  background: rgb(34, 193, 195);
  background: linear-gradient(
    90deg,
    rgba(34, 193, 195, 1) 0%,
    rgba(253, 187, 45, 1) 100%
  );
  border: 0px;
  cursor: pointer;
  transition: opacity 0.25s ease-out;
}

.row button:hover {
  opacity: 0.8;
}

#button {
  padding-bottom: 1.5rem;
}

#iconGroup {
  display: flex;
  justify-content: center;
  padding: 2rem 0;
}

#iconGroup a {
  margin: 0 0.6rem;
}

#alternativeLogin {
  text-align: center;
  padding-top: 2rem;
  margin-top: 1.5rem;
}
