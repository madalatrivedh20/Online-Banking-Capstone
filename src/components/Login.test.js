import { render, screen, waitForElementToBeRemoved } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Login from './Login';

import {
  BrowserRouter as Router,
  Routes,
  Route
} from "react-router-dom";
import { AppStateProvider } from '../AppStateContext';

it('Snapshot Test', () => {
  const component = renderer.create(
    <AppStateProvider>
      <Router>
        <Login />
      </Router>
    </AppStateProvider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders the page', async () => {
  render(
    <AppStateProvider>
      <Router>
        <Login />
      </Router>
    </AppStateProvider>
  );

  expect(screen.getByRole("heading")).toHaveTextContent(/Login/);
  expect(screen.getByPlaceholderText("Enter your Email")).toBeInTheDocument();
  expect(screen.getByPlaceholderText("Enter your Password")).toBeInTheDocument();
  expect(screen.getByRole("button", { name: "Log In" })).toBeInTheDocument();
});