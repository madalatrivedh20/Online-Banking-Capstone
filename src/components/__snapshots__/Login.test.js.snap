// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 1`] = `
<div
  className="forms"
>
  <div>
    <div
      className="imgcontainer"
    >
      <img
        alt="Avatar"
        className="avatar"
        src="download.png"
      />
    </div>
    <h2
      id="headerTitle"
    >
      Login
    </h2>
  </div>
  <div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="row"
      >
        <label>
          Email
        </label>
        <input
          name="email"
          onChange={[Function]}
          placeholder="Enter your Email"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="row"
      >
        <label>
          Password
        </label>
        <input
          name="password"
          onChange={[Function]}
          placeholder="Enter your Password"
          required={true}
          type="password"
          value=""
        />
      </div>
      <div
        className="row"
        id="button"
      >
        <button>
          Log In
        </button>
      </div>
      <hr />
      <div
        className="row"
        id="button"
      >
        <div
          style={
            Object {
              "marginLeft": "0px",
              "textAlign": "center",
            }
          }
        >
          <ul>
            <li
              style={
                Object {
                  "display": "inline",
                }
              }
            >
               
              <a
                class="fa-brands fa-facebook"
                href="#"
                onClick={[Function]}
              />
            </li>
            <li
              style={
                Object {
                  "display": "inline",
                }
              }
            >
               
              <a
                class="fa-brands fa-google"
                href="#"
                onClick={[Function]}
              />
            </li>
            <li
              style={
                Object {
                  "display": "inline",
                }
              }
            >
               
              <a
                class="fa-brands fa-twitter"
                href="#"
                onClick={[Function]}
              />
            </li>
          </ul>
        </div>
      </div>
    </form>
  </div>
  <div
    className="container"
    style={
      Object {
        "backgroundColor": "#f1f1f1",
        "height": "50px",
        "textAlign": "center",
      }
    }
  >
    New User ? 
    <a
      href="/register"
    >
      Register Here
    </a>
  </div>
</div>
`;
